drivers/bridge/bridge.go

```diff
@@ -1060,16 +1056,26 @@ func (d *driver) CreateEndpoint(nid, eid string, epInfo driverapi.EndpointInfo,
 		endpoint.addrv6 = ipv6Addr
 	}
 
-	err = epInfo.AddInterface(endpoint.macAddress, *ipv4Addr, *ipv6Addr)
+	// Program any required port mapping and store them in the endpoint
+	endpoint.portMapping, err = n.allocatePorts(epConfig, endpoint, config.DefaultBindingIP, d.config.EnableUserlandProxy)
 	if err != nil {
 		return err
 	}
 
-	// Program any required port mapping and store them in the endpoint
-	endpoint.portMapping, err = n.allocatePorts(epConfig, endpoint, config.DefaultBindingIP, d.config.EnableUserlandProxy)
+	err = ifInfo.SetMacAddress(endpoint.macAddress)
```

Comment:
It is not related to this PR. But just as a point to note that we must handle the case of user setting mac-address using the --mac-address option. That belongs to the PR to integrate IPAM with bridge driver.